src/webgpu/shader/validation/parse/literal.spec.ts

```diff
@@ -0,0 +1,137 @@
+export const description = `Validation tests for literals`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+g.test('bools')
+  .desc(`Test that valid bools are accepted.`)
+  .params(u => u.combine('val', ['true', 'false']).beginSubcases())
+  .fn(t => {
+    const code = `var test = ${t.params.val};`;
+    t.expectCompileResult(true, t.wrapInEntryPoint(code));
+  });
+
+{
+  const kValidIntegers = new Set([
+    '0x123', // hex number
+    '123', // signed number, no suffix
+    '94i', // signed number
+    '1u', // unsigned number
+    '0', // zero
+    '0x3f', // hex with 'f' as last character
+    '2147483647', // max signed int
+    '2147483647i', // max signed int
+    '2147483647u', // max signed int
+    '-2147483648', // min signed int
+    '-2147483648i', // min signed int
+    '4294967295', // will be deduced to unsigned
```

Comment:
I don't think this one is right. Without a suffix, I think this gets deduced as an i32, and will therefore be an error. See the `i32_too_large_2` example in the spec.

Reply:
Hm, why doesn't the rank 4 AbstractInt -> u32 kick in?